baybe/surrogates/gaussian_process/core.py

```diff
@@ -146,3 +147,11 @@ def _fit(self, searchspace: SearchSpace, train_x: Tensor, train_y: Tensor) -> No
             botorch.optim.fit.fit_gpytorch_mll_torch(mll, step_limit=200)
         else:
             botorch.fit.fit_gpytorch_mll(mll)
+
+    def __str__(self) -> str:
```

Comment:
I'm glad that finally someone is taking care of the printing, because I really wasn't happy with the indentation and a few other things in our first version 👍🏼 However, since you've already decided to give it a shot, I suggest we follow a consistent approach from the get-go. 

What still bothers me with the current version: you've simply fixed it for GPs, but as soon as I use a random forest / some other surrogate, everything's still messed up. Why not do it properly and:
* Define printing of base-class attributes directly in the base class and then only ...
* ... add the subclass fields in the corresponding subclass while properly taking into account indentation?

Reply:
Will adjust.